themes/default/content/blog/next-level-iac-package-ecosystems/index.md

```diff
@@ -60,26 +64,95 @@ export const bucketName = bucket.id;
 
 // Export the URL of the file in the S3 bucket
 export const objectUrl = pulumi.interpolate`s3://${bucket.id}/${bucketObject.key}`;
-``` 
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+import pulumi
+import pulumi_aws as aws
+
+# Create an S3 bucket
+bucket = aws.s3.Bucket('my-bucket')
+
+# The path to the file to upload to S3
+file_path = "path/to/your/local/file.ext"  # Replace with the path to the file you want to upload
+
+# Read the contents of the file to be uploaded to S3.
+with open(file_path, 'rb') as file:
```

Comment:
When I run this, I get an error. Seems like it should be this (which results in a `str`)?

```suggestion
with open(file_path) as file:
```

Reply:
Hmm.. Yeah, I just assumed the `BucketObject.content` wanted binary data. Looking over it, apparently it wants.. text/string? Strange API choice, TBH. So however you can use `source` which takes binary. I noticed that most of the examples out there for Python actually just use `source=pulumi.FileAsset('./file.txt')` rather than using the `content` property. So maybe I'll update both the Typescript and the Python examples to use `FileAsset` instead. I was mostly just working from the [example given in the AI Answers page](https://www.pulumi.com/ai/answers/fZ6JumwyAXHcXys5tFFkqq/uploading-files-to-amazon-s3-using-typescript), which uses `content`.  Realizing now that is probably not best practice, so...